Lib/pickle.py

```diff
@@ -1035,12 +1035,11 @@ def _batch_appends(self, items, obj):
 
         it = iter(items)
         start = 0
-        while True:
-            tmp = list(islice(it, self._BATCHSIZE))
-            n = len(tmp)
+        for batch in batched(it, self._BATCHSIZE):
+            n = len(batch)
             if n > 1:
```

Comment:
You can inline `n` which is only used one time. Also, it would be safer to write the condition as `len(batch) != 1`.